src/hardware/mod.rs

```diff
@@ -44,33 +49,55 @@ pub trait Reset {
 pub(crate) mod fake {
     use core::convert::TryInto;
     use core::time::Duration;
+    use std::collections::HashMap;
 
     /// A fake `Identity` that returns fixed values.
     pub struct Identity {
         firmware_version: Vec<u8>,
+        vendor_firmware_versions: HashMap<u8, Vec<u8>>,
         unique_id: Vec<u8>,
     }
 
     impl Identity {
         /// Creates a new `fake::Identity`.
-        pub fn new(firmware_version: &[u8], unique_id: &[u8]) -> Self {
-            let mut firmware_version = firmware_version.to_vec();
-            while firmware_version.len() < 32 {
-                firmware_version.push(0);
+        pub fn new(
+            firmware_version: &[u8],
+            vendor_firmware_versions: HashMap<u8, &[u8]>,
+            unique_id: &[u8],
+        ) -> Self {
+            let mut vendor_fw_versions_vec = HashMap::new();
+            for (key, val) in vendor_firmware_versions.iter() {
+                vendor_fw_versions_vec.insert(*key, Self::u8_to_sized_vec(val));
             }
-            firmware_version.truncate(32);
 
             Self {
-                firmware_version,
+                firmware_version: Self::u8_to_sized_vec(firmware_version),
+                vendor_firmware_versions: vendor_fw_versions_vec,
                 unique_id: unique_id.to_vec(),
             }
         }
+
+        fn u8_to_sized_vec(data: &[u8]) -> Vec<u8> {
```

Comment:
I think this makes more sense as a local function in `new()`. WDYT? Also, perhaps a name like `pad_to_32` is more appropriate?

Reply:
Done. I actually didn't know you could have local `fn`s.